ctru-rs/src/services/soc.rs

```diff
@@ -1,18 +1,18 @@
-use libc::{free, memalign};
+use libc::memalign;
+use std::lazy::SyncLazy;
 use std::net::Ipv4Addr;
-use std::sync::atomic::{AtomicBool, Ordering};
+use std::sync::Mutex;
 
-use crate::error::Error;
-use ctru_sys::{socExit, socInit};
+use crate::services::ServiceHandler;
 
 /// Soc service. Initializing this service will enable the use of network sockets and utilities
 /// such as those found in `std::net`. The service will be closed when this struct is is dropped.
 #[non_exhaustive]
 pub struct Soc {
-    soc_mem: *mut u32,
```

Comment:
We should still deallocate the memory. To store the memory address, you might need another Mutex.

Reply:
nvm, turns out we lose access to this memory when we pass it in.